### PR TITLE
Vector input

### LINKED ISSUE #216: Vector input
For certain contracts, for example Multisig, a vector input is needed. The user should be able to add and remove inputs for the vector elements.
This is how it looks in Apps

Screenshot 2022-02-25 at 17 26 34

### PR DESCRIPTION
**Pull request by achimcc**
Allows for `Vector` input as parameters for constructors and function calls, like `Vec<AccountId>` in the [multisig](https://github.com/paritytech/ink/tree/master/examples/multisig) example contract.

Closes #216

### CODE REVIEW

**Comment by achimcc** [comment]
I did test the deployment and functionalities of the multisig contract with this PR.

I wrote an ink! contract which receives a `Vec<Vec<Vec<AccountId>>>` as input for testing purposes:

https://github.com/achimcc/multisig_nested_vec_input/

Looks currently like this:

image


However, I'd sugggest to merge PR #225 first, I'd rebnase faterwards and talk to @kwingram25 about the usage of the registry. I currently don't see the need that we have to use it, it would probably rather be an improvement?

File: src/ui/components/form/Vector.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  const _onChange = (row: number) => (newEntry: unknown) => {
+    const newValue = value.map((entry, index) => (index == row ? newEntry : entry));
+    onChange(newValue);
+  };
+  const isOddNesting = nestingNumber % 2 != 0;
+  return (
+    <div
+      className={`p-4 text-left text-sm ${
+        isOddNesting ? 'dark:bg-gray-900 bg-white' : 'dark:bg-gray-500 bg-gray-100'
+      } rounded border dark:border-gray-500 border-gray-200`}
```
Comment (statictype) [inline src/ui/components/form/Vector.tsx:55]: `dark:border-gray-700` here would be more in line with the other inputs

File: src/ui/components/form/Vector.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  const _rowRemove = () => onChange(value.slice(0, -1));
+  const _onChange = (row: number) => (newEntry: unknown) => {
+    const newValue = value.map((entry, index) => (index == row ? newEntry : entry));
+    onChange(newValue);
+  };
+  const isOddNesting = nestingNumber % 2 != 0;
+  return (
+    <div
+      className={`p-4 text-left text-sm ${
+        isOddNesting ? 'dark:bg-gray-900 bg-white' : 'dark:bg-gray-500 bg-gray-100'
```
Comment (statictype) [inline src/ui/components/form/Vector.tsx:54]: `dark:bg-elevation-1` instead of `dark:bg-gray-500` matches the sidebar color and the mock-up

**Review by statictype — COMMENTED** [review]
Maybe we can also change the placeholder of the `AddressSelect` and `Select` components to say something like 'Select an account' instead of 'No Addresses Found'. 
Ideally the first dropdown option should be selected by default, but might be too much for this PR

File: src/ui/components/form/Vector.tsx (hunk 2 above)
Comment (achimcc) [reply src/ui/components/form/Vector.tsx:54]: True, looks much better! Done!

**Comment by achimcc** [comment]
> Maybe we can also change the placeholder of the `AddressSelect` and `Select` components to say something like 'Select an account' instead of 'No Addresses Found'. Ideally the first dropdown option should be selected by default, but might be too much for this PR

I agree, but I think this is where also the registry comes into play, and we can probably do this in a later PR. Allowing form inputs  (Issue #232 ) might also be higher priority, since this is missing to become feature complete in the sense of being able to use all the ink! examples in Contracts UI.

**Review by statictype — APPROVED** [review]

**Pull request merged**